docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1232 @@
+        await interaction.response.send_message(to_send)
+
+When referring to a renamed parameter in other decorators, the original parameter name should be used.
+For example, to use :func:`~.app_commands.describe` and :func:`~.app_commands.rename` together:
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.describe(
+        content="the text to repeat",
+        n_times="the number of times to repeat the text"
+    )
+    @app_commands.rename(n_times="number-of-times")
+    async def repeat(interaction: discord.Interaction, content: str, n_times: int):
+        to_send = textwrap.shorten(f"{content} " * n_times, width=2000)
+        await interaction.response.send_message(to_send)
+
+Choices
+++++++++
+
+To provide the user with a list of options to choose from for an argument, the :func:`.app_commands.choices` decorator can be applied.
+
+A user is restricted to selecting a choice and can't type something else.
+
+Each individual choice contains 2 fields:
+
+- A name, which is what the user sees
+- A value, which is hidden to the user and only visible to the API. Typically, this is either the same as the name or something more developer-friendly. Value types are limited to either a :class:`str`, :class:`int` or :class:`float`.
+
+To illustrate, the following command has a selection of 3 colours with each value being the colour code:
+
+.. code-block:: python
+
+    @client.tree.command()
+    @app_commands.describe(colour="pick your favourite colour")
+    @app_commands.choices(colour=[
+        app_commands.Choice(name="Red", value=0xFF0000),
+        app_commands.Choice(name="Green", value=0x00FF00),
+        app_commands.Choice(name="Blue", value=0x0000FF)
+    ])
+    async def colour(interaction: discord.Interaction, colour: app_commands.Choice[int]):
+        """show a colour"""
+
+        embed = discord.Embed(title=colour.name, colour=colour.value)
+        await interaction.response.send_message(embed=embed)
+
+On the client:
+
+.. image:: /images/guide/app_commands/colour_command_preview.png
+    :width: 400
+
+discord.py also supports 2 other pythonic ways of adding choices to a command,
+shown :func:`here <discord.app_commands.choices>` in the reference.
+
+Autocompletion
++++++++++++++++
+
+Autocomplete callbacks allow the bot to dynamically return up to 25 choices
+to a user as they type a parameter.
+
+In short:
+
+- User starts typing.
+
+- After a brief debounced pause from typing, Discord requests a list of choices from the bot.
+
+- An autocomplete callback is called with the current user input.
+
+- Returned choices are sent back to Discord and shown in the user's client.
+
+  - An empty list can be returned to denote no choices.
+
+Attaching an autocomplete function to a parameter can be done in 2 main ways:
+
+1. From the command, with the :meth:`~.app_commands.Command.autocomplete` decorator
+2. With a separate decorator, :func:`.app_commands.autocomplete`
```

Comment:
It is also possible to attach an autocompletion via a [Transformer](https://discordpy.readthedocs.io/en/latest/interactions/api.html?highlight=transformer%20autocomplete#discord.app_commands.Transformer.autocomplete)